src/Core/FeatureFlag/Layer/DotEnvLayer.php

```diff
@@ -0,0 +1,149 @@
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\FeatureFlag\Layer;
+
+use PrestaShop\PrestaShop\Core\ConfigurationInterface;
+use PrestaShop\PrestaShop\Core\EnvironmentInterface;
+use PrestaShop\PrestaShop\Core\FeatureFlag\FeatureFlagSettings;
+use PrestaShop\PrestaShop\Core\FeatureFlag\TypeLayerInterface;
+
+class DotEnvLayer implements TypeLayerInterface
+{
+    public function __construct(
+        private EnvironmentInterface $environment,
+        private ConfigurationInterface $configuration
+    ) {
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isReadonly(): bool
+    {
+        // It's always editable via DotEnv layer!
+        return false;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getTypeName(): string
+    {
+        return FeatureFlagSettings::TYPE_DOTENV;
+    }
+
+    /**
+     * Retrieve the variable name of this feature flag.
+     */
+    public function getVarName(string $featureFlagName): string
+    {
+        return FeatureFlagSettings::PREFIX . strtoupper($featureFlagName);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function canBeUsed(string $featureFlagName): bool
+    {
+        return isset($_ENV['SYMFONY_DOTENV_VARS']) &&
+            str_contains($_ENV['SYMFONY_DOTENV_VARS'], $this->getVarName($featureFlagName));
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isEnabled(string $featureFlagName): bool
+    {
+        return isset($_ENV[$this->getVarName($featureFlagName)]) &&
+            filter_var($_ENV[$this->getVarName($featureFlagName)], \FILTER_VALIDATE_BOOLEAN);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function enable(string $featureFlagName): void
+    {
+        $this->setStatus($featureFlagName, true);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function disable(string $featureFlagName): void
+    {
+        $this->setStatus($featureFlagName, false);
+    }
+
+    /**
+     * Retrieve dotenv file used with this feature flag set.
+     */
+    private function locateDotEnvFile(string $featureFlagName): string
+    {
+        $env = $this->environment->getName();
+        $filesToCheck = [".env.$env.local", ".env.$env", '.env'];
+
+        foreach ($filesToCheck as $file) {
+            $path = $this->configuration->get('_PS_ROOT_DIR_') . '/' . $file;
```

Comment:
This should be injected in the constructor and in the DI you can inject the const